### PR TITLE
[DisplayServer] Add method to check if window transparency is supported and enabled.

### PR DESCRIPTION
**Pull request by bruvzg**
Adds method to check is transparency is usable, see https://github.com/godotengine/godot/pull/91333

### CODE REVIEW

File: doc/classes/DisplayServer.xml [hunk 1]
```diff
@@ -930,6 +930,12 @@
 				Returns [code]true[/code] if touch events are available (Android or iOS), the capability is detected on the Web platform or if [member ProjectSettings.input_devices/pointing/emulate_touch_from_mouse] is [code]true[/code].
 			</description>
 		</method>
+		<method name="is_window_transparency_enabled" qualifiers="const">
+			<return type="bool" />
+			<description>
+				Returns [code]true[/code] if window background can be set transparent. This method return [code]false[/code] if [member ProjectSettings.display/window/per_pixel_transparency/allowed] is set to [code]false[/code], or transparency is not supported by renderer, or OS compositor.
```
Comment (akien-mga) [inline doc/classes/DisplayServer.xml:936]:
```suggestion
				Returns [code]true[/code] if the window background can be set transparent. This method return [code]false[/code] if [member ProjectSettings.display/window/per_pixel_transparency/allowed] is set to [code]false[/code], or if transparency is not supported by the renderer or OS compositor.
```

**Review by akien-mga — APPROVED** [review]

**Review by AThousandShips — APPROVED** [review]

**Comment by LunarTides** [comment]
To me, "is_window_transparency_enabled" reads like "is_window_transparency_**currently**_enabled" which might cause confusion. Maybe rename it to something like "is_window_transparency_supported"?

**Comment by LunarTides** [comment]
> I guess that, in a way, it _is_ "currently" enabled.

I am not exactly sure how the display server handles transparency, but based on its name and description in the documentation: _"Returns true if the window background can be made transparent"_, it seems like more of a "is this feature supported/available" rather than "is this feature enabled". Or at least it would seem that way had I looked at its documentation, although I may be misunderstanding what this function is for, or maybe the phrasing is just throwing me off.

**Comment by akien-mga** [comment]
Thanks!

**Comment by RandomShaper** [comment]
I know I'm late but I'd like to support the idea of this being checked via `has_feature(FEATURE_WINDOW_TRANSPARENCY)`.

I'm aware the new function added here checks if both supported and enabled. However, that's something that projects could check anyway via `has_feature()` and the `display/window/per_pixel_transparency/allowed` project setting.

If this new function is really important as a shortcut (which I'm not very sure about), it could be implemented in such terms.

**Comment by RandomShaper** [comment]
That's indeed how it's being used at the moment for the most part. However, the method is not described as not able to perform checks at runtime. `DisplayServerWeb` is an example of something we already have that isn't following such a constraint, by making runtime checks for features such as `FEATURE_IME`.

We could maybe specify the method as being runtime-constant (instead of compile-time constant).

**Pull request merged**